### PR TITLE
19163 & 19165 - Credits implementation to assist customers where their credits have been spent

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*
https://github.com/bcgov/entity/issues/19163
https://github.com/bcgov/entity/issues/19165

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-queue/src/pay_queue/services/payment_reconciliations.py [hunk 1]
```diff
@@ -568,6 +616,13 @@ def _create_credit_records(csv_content: str):
                     account_id=pay_account.id
                 ).save()
 
+    for row in csv.DictReader(csv_content.splitlines()):
```
Comment (seeker25) [inline pay-queue/src/pay_queue/services/payment_reconciliations.py:619]: Needs to be done after the credits have been established

File: pay-queue/src/pay_queue/services/payment_reconciliations.py [hunk 2]
```diff
@@ -395,6 +401,48 @@ def _csv_error_handling(row, error_msg: str, error_messages: List[Dict[str, any]
...
+        cfs_identifier=cfs_identifier,
+        created_on=datetime.strptime(_get_row_value(row, Column.APP_DATE), '%d-%b-%y'),
+        credit_id=credit.id,
+        invoice_amount=Decimal(_get_row_value(row, Column.TARGET_TXN_ORIGINAL)),
+        invoice_number=invoice_number
+    ).flush()
+
+    amount = CfsCreditInvoices.credit_for_invoice_number(invoice_number)
+    invoices = db.session.query(InvoiceModel) \
+        .join(InvoiceReferenceModel, InvoiceReferenceModel.invoice_id == InvoiceModel.id) \
```
Comment (ochiu) [inline pay-queue/src/pay_queue/services/payment_reconciliations.py:429]: is the invoice reference status relevant / needed here?

Comment (seeker25) [reply pay-queue/src/pay_queue/services/payment_reconciliations.py:429]: There wouldn't be a row without it being a completed invoice reference and we're only looking at PAID invoices

Comment (seeker25) [reply pay-queue/src/pay_queue/services/payment_reconciliations.py:429]: put in two checks just in case

**Review by ochiu — APPROVED** [review]

**Pull request merged**